### PR TITLE
Pdfng reload on error

### PR DESCRIPTION
**Pull request by briangough**
Catch errors from pdfjs where possible, and use a 60 second timeout to kill pdfjs and force reload of document if page does not load or render.

If document has to be reloaded more than twice, show the "Compile failed message" so user has to click recompile manually.

### CODE REVIEW

File: public/coffee/ide/pdfng/directives/pdfRenderer.coffee [hunk 1]
```diff
@@ -57,15 +62,20 @@ define [
...
+				, (error) ->
+					console.log 'ERROR', error
 
-				@document.then (pdfDocument) ->
-					pdfDocument.getDestination(dest)
-
+				# @document.then (pdfDocument) ->
+				# 	pdfDocument.getDestination(dest)
+				# , (error) ->
+				# 	console.log 'ERROR', error
```
Comment (jpallen) [inline public/coffee/ide/pdfng/directives/pdfRenderer.coffee]: Is this code dead now? If so we should remove it fully I think.

File: public/coffee/ide/pdfng/directives/pdfRenderer.coffee [hunk 2]
```diff
@@ -118,17 +129,38 @@ define [
...
+				timedOut = false
+				timer = $timeout () =>
+					Raven.captureMessage?('pdfng page load timed out after ' + @PAGE_LOAD_TIMEOUT + 'ms')
+					# console.log 'page load timed out', pagenum
+					timedOut = true
+					@spinner.stop(element.canvas)
+					# @jobs = @jobs - 1
+					# @triggerRenderQueue(0)
+					this.errorCallback?('timeout')
+				, @PAGE_LOAD_TIMEOUT
```
Comment (jpallen) [inline public/coffee/ide/pdfng/directives/pdfRenderer.coffee]: Do we need to do anything to stop the render that is potentially still going on in the background? If not, why not?

**Comment by jpallen** [comment]
No obvious issues from just reading the code, but I don't fully understand the context of some of the changes. If it's tested on your end, I'd say it's good to go.

File: public/coffee/ide/pdfng/directives/pdfRenderer.coffee (hunk 2 above)
Comment (briangough) [inline public/coffee/ide/pdfng/directives/pdfRenderer.coffee]: On successful page load the timeout is immediately cancelled before any rendering is started so this should be ok as only one page is rendered at any time.

File: public/coffee/ide/pdfng/directives/pdfRenderer.coffee (hunk 1 above)
Comment (briangough) [inline public/coffee/ide/pdfng/directives/pdfRenderer.coffee]: The commented code will be what we use when we upgrade to a newer version of pdfjs with a direct method for getDestination, I'll change the comment to make that clearer.

**Pull request merged**